### PR TITLE
feat: update kingfisher process URLs and use metadata endpoint

### LINKED ISSUE #345: Move update_collection_availability() to Pelican task
The process manager shouldn't know the details of individual tasks. Presently, update_collection_availability() violates this.

On the other hand, tasks shouldn't know too much about the job. But, tasks already know a fair bit about the job (setting `context` used by other jobs). So, although there's not a clean separation of concerns, it's more consistent to move the `update_collection_availability` logic into the Pelican task's `get_status` method.

### PR DESCRIPTION
**Pull request by yolile**
ref #291 
closes #345

### CODE REVIEW

File: data_registry/process_manager/task/process.py [hunk 1]
```diff
@@ -38,6 +40,20 @@ def get_status(self):
             self.job.context["process_data_version"] = compile_releases.get("data_version")
             self.job.save()
 
+        if is_last_completed:
+            response = request(
+                "GET",
+                urljoin(settings.KINGFISHER_PROCESS_URL, f"{base_url}/{self.process_id}/metadata/"),
```
Comment (jpmckinney) [inline data_registry/process_manager/task/process.py:46]: Hmm, you wrote this endpoint to accept the compiled collection, rather than the base collection. The compiled collection is stored in `compile_releases`.

File: data_registry/process_manager/task/process.py [hunk 2]
```diff
@@ -8,6 +8,8 @@
 
 logger = logging.getLogger(__name__)
 
+base_url = "/api/collections"
```
Comment (jpmckinney) [inline data_registry/process_manager/task/process.py:11]: Instead of extracting only the base URL, can do:

```python
def _url(path):
    return urljoin(settings.KINGFISHER_PROCESS_URL, f"/api/collections/{path}")
```

File: data_registry/process_manager/task/process.py (hunk 1 above)
Comment (jpmckinney) [reply data_registry/process_manager/task/process.py]: Indeed. `process_id` is set to the original collection by the collect task (extracted from the crawl log). Then `process_id_pelican` is set to the compiled collection by the process task (because Pelican operates on compiled releases).

**Comment by jpmckinney** [comment]
Ah, because of my refactoring we'll also have to do https://github.com/open-contracting/data-registry/issues/345 as part of this.

**Comment by jpmckinney** [comment]
I have to switch tasks for a bit, so let me know if you want to do that @yolile.

**Comment by yolile** [comment]
> I have to switch tasks for a bit, so let me know if you want to do that @yolile.

Sure, If you have pushed all your changes I can take it from here.

**Comment by jpmckinney** [comment]
All pushed now

File: data_registry/process_manager/task/process.py [hunk 3]
```diff
@@ -16,7 +16,7 @@ def url_for_collection(*parts):
 class Process(TaskManager):
     def __init__(self, job):
         self.job = job
-        self.process_id = job.context["process_id"]
+        self.process_id = job.context.get("process_id", None)
```
Comment (jpmckinney) [inline data_registry/process_manager/task/process.py:19]: Why was this change needed?

File: data_registry/process_manager/task/pelican.py [hunk 4]
```diff
@@ -64,6 +65,8 @@ def get_pelican_id(self):
         return self.job.context.get("pelican_id")
 
     def get_pelican_dataset_name(self):
+        if "process_data_version" not in self.job.context:
+            raise RecoverableException("Process data version is not set")
```
Comment (jpmckinney) [inline data_registry/process_manager/task/pelican.py:69]: I don't think this scenario is actually recoverable?

File: data_registry/process_manager/task/process.py (hunk 3 above)
Comment (jpmckinney) [reply data_registry/process_manager/task/process.py:19]: Aha, right, when these classes are initialized only to run wipe(). I'll see if there's a better way.

Comment (jpmckinney) [reply data_registry/process_manager/task/process.py]: Okay, I think I fixed that logic issue correctly now. Can you test? (Ideally we would have an automated test as well)

Comment (yolile) [reply data_registry/process_manager/task/process.py]: Now it works

**Review by jpmckinney — APPROVED** [review]
Phew, should be easier to make changes to this next time

**Comment by yolile** [comment]
What a refactoring! The code looks more consistent and easier to maintain indeed!
